pkg/commands/componentarchive/remote/remote_test.go

```diff
@@ -0,0 +1,118 @@
+// SPDX-FileCopyrightText: 2021 SAP SE or an SAP affiliate company and Gardener contributors.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package remote_test
+
+import (
+	"context"
+	"os"
+
+	"github.com/go-logr/logr"
+	"github.com/mandelsoft/vfs/pkg/layerfs"
+	"github.com/mandelsoft/vfs/pkg/memoryfs"
+	"github.com/mandelsoft/vfs/pkg/osfs"
+	"github.com/mandelsoft/vfs/pkg/projectionfs"
+	"github.com/mandelsoft/vfs/pkg/vfs"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"github.com/gardener/component-cli/ociclient/options"
+	"github.com/gardener/component-cli/pkg/commands/componentarchive/remote"
+)
+
+var _ = Describe("Remote", func() {
+
+	var testdataFs vfs.FileSystem
+
+	BeforeEach(func() {
+		baseFs, err := projectionfs.New(osfs.New(), "../")
+		Expect(err).ToNot(HaveOccurred())
+		testdataFs = layerfs.New(memoryfs.New(), baseFs)
+	})
+
+	It("should push a component archive", func() {
+		baseFs, err := projectionfs.New(osfs.New(), "../")
+		Expect(err).ToNot(HaveOccurred())
+		testdataFs = layerfs.New(memoryfs.New(), baseFs)
+		ctx := context.Background()
+
+		cf, err := testenv.GetConfigFileBytes()
+		Expect(err).ToNot(HaveOccurred())
+		Expect(vfs.WriteFile(testdataFs, "/auth.json", cf, os.ModePerm))
+
+		pushOpts := &remote.PushOptions{
+			OciOptions: options.Options{
+				AllowPlainHttp:     false,
+				RegistryConfigPath: "/auth.json",
+			},
+		}
+		pushOpts.ComponentArchivePath = "./testdata/00-ca"
+		pushOpts.BaseUrl = testenv.Addr + "/test"
+
+		Expect(pushOpts.Run(ctx, logr.Discard(), testdataFs)).To(Succeed())
+
+		res := remote.NewPushCommand(ctx)
+		Expect(res)
+
+		repos, err := client.ListRepositories(ctx, testenv.Addr+"/test")
+		Expect(err).ToNot(HaveOccurred())
+
+		expectedRef := testenv.Addr + "/test/component-descriptors/example.com/component"
+		Expect(repos).To(ContainElement(Equal(expectedRef)))
+
+	})
+
+	It("should get component archive", func() {
```

Comment:
this test can only work if the test `should push a component archive` was running before.
They they have to run in sequence which is not guaranteed.

Can you update the test so that it can work idependant.